include/flx/forward_automatic/gradient.hpp

```diff
@@ -0,0 +1,163 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/overload.hpp>
+#include <flx/forward_automatic/var.hpp>
+#include <flx/forward_automatic/valder.hpp>
+#include <flx/forward_automatic/val.hpp>
+#include <flx/forward_automatic/der.hpp>
+
+namespace eve
+{
+
+  //================================================================================================
+  //================================================================================================
+  // Function decorators mark-up used in function overloads
+  struct pol_
+  {
+    template<typename D> static constexpr auto combine(D const&) noexcept = delete;
+  };
+
+  using pol_type = decorated<pol_()>;
+  //================================================================================================
+  //! @addtogroup core_decorators
+  //! @{
+  //! @var pol
+  //!
+  //! @brief  Higher-order @callable imbuing polar semantic onto other @callable{s}.
+  //!
+  //! #### Members Functions
+  //!
+  //!  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.cpp}
+  //!  auto operator()(eve::callable auto const& f ) const noexcept;
+  //!  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  //! @param f
+  //! An instance of eve::callable
+  //!
+  //! @return
+  //! A @callable performing the same kind of operation but implying polar semantic.
+  //!
+  //!  @}
+  //================================================================================================
+  [[maybe_unused]] inline constexpr pol_type const pol = {};
+
+
+  EVE_REGISTER_CALLABLE(gradient_);
+  EVE_DECLARE_CALLABLE(gradient_,gradient);
+  EVE_CALLABLE_API(gradient_,gradient);
+}
+
+
+namespace flx
+{
+  inline eve::detail::callable_object<eve::tag::gradient_> const gradient = {};
+}
+
+namespace eve
+{
+  //================================================================================================
```

Comment:
This needs to be put before `gradient = {};` or it will never be picked up by doxygen

Reply:
ok

Reply:
why is it in a eve namespace ??